src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -88,8 +87,10 @@ public Task<Stream> StartCpuTrace(int pid, int durationSeconds)
                 }
             });
 
-            return Task.FromResult(session.EventStream);
+            //Always cleanup linked token source.
+            traceTask.ContinueWith((_) => linkedTokenSource.Dispose());
```

Comment:
Ew continue with 😢

Reply:
I had the same reaction, but I do not know of a better alternative :(

---
In reply to: [405171622](https://github.com/dotnet/diagnostics/pull/979#discussion_r405171622) [](ancestors = 405171622)

Reply:
Why not put it inside of the Task.Run after session.Stop?

Reply:
In this particular case, I think that would be ok, but it's a bit of an anti-pattern: I really want to express that the disposal happens regardless of what happens in the body of the task, or even if the task is never executed (because someone added a cancellation token to Task.Run).

---
In reply to: [405220914](https://github.com/dotnet/diagnostics/pull/979#discussion_r405220914) [](ancestors = 405220914)

Reply:
Ya but you own the task.